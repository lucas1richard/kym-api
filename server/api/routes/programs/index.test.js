const { expect, use } = require('chai');
const { jestSnapshotPlugin } = require('mocha-chai-jest-snapshot');

describe('programs routes', () => {
  use(jestSnapshotPlugin());
  const routePrefix = '/api/programs';

  describe(`GET ${routePrefix}/v1`, () => {
    before(globals.seedTestData);
    after(globals.destroyAllHook);

    it('gets all programs for the user', async () => {
      const res = await globals.agent.getWithToken(`${routePrefix}/v1`);
      expect(res.body).toMatchSnapshot();
    });
  });

  describe(`GET ${routePrefix}/single-program/v1`, () => {
    before(globals.seedTestData);
    after(globals.destroyAllHook);

    it('gets a single program for the user', async () => {
      const res = await globals.agent.getWithToken(`${routePrefix}/single-program/v1`);
      expect(res.body).toMatchSnapshot();
    });
  });

  describe(`POST ${routePrefix}/v1`, () => {
    before(globals.seedTestData);
    after(globals.destroyAllHook);

    it('creates a program for the user', async () => {
      const res = await globals.agent.postWithToken(`${routePrefix}/v1`)
        .send({ units: 'IMPERIAL', weight: 175, goal: 'GAIN_MUSCLE' });

      expect(res.statusCode).equal(201);
      expect(res.body.id).to.be.ok;
    });

    it('validates the req', async () => {
      const res = await globals.agent.postWithToken(`${routePrefix}/v1`)
        .send({});

      expect(res.statusCode).equal(400);
      expect(res.body).toMatchSnapshot();
    });
  });
});
