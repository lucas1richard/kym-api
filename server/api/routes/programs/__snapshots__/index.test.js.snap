// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`programs routes GET /api/programs/single-program/v1 gets a single program for the user 1`] = `
Object {
  "enddate": "2022-10-23",
  "endgoal": "180",
  "endweight": null,
  "id": 1,
  "result": "TBD",
  "startdate": "2022-09-18",
  "startweight": "175",
  "status": "IN_PROGRESS",
  "user_uuid": "60f24265-ef45-4891-b4ef-e8fad4435308",
}
`;

exports[`programs routes GET /api/programs/v1 gets all programs for the user 1`] = `
Array [
  Object {
    "enddate": "2022-10-23",
    "endgoal": "180",
    "endweight": null,
    "id": 1,
    "result": "TBD",
    "startdate": "2022-09-18",
    "startweight": "175",
    "status": "IN_PROGRESS",
    "user_uuid": "60f24265-ef45-4891-b4ef-e8fad4435308",
  },
]
`;

exports[`programs routes POST /api/programs/v1 creates a program for the user 1`] = `
Object {
  "enddate": "2022-10-23",
  "endgoal": "180",
  "endweight": null,
  "id": 6037,
  "result": "TBD",
  "startdate": "2022-09-18",
  "startweight": "175",
  "status": "IN_PROGRESS",
  "user_uuid": "60f24265-ef45-4891-b4ef-e8fad4435308",
}
`;

exports[`programs routes POST /api/programs/v1 validates the req 1`] = `
Array [
  Object {
    "context": Object {
      "key": "units",
      "label": "units",
    },
    "message": "\\"units\\" is required",
    "path": Array [
      "units",
    ],
    "type": "any.required",
  },
  Object {
    "context": Object {
      "key": "weight",
      "label": "weight",
    },
    "message": "\\"weight\\" is required",
    "path": Array [
      "weight",
    ],
    "type": "any.required",
  },
  Object {
    "context": Object {
      "key": "goal",
      "label": "goal",
    },
    "message": "\\"goal\\" is required",
    "path": Array [
      "goal",
    ],
    "type": "any.required",
  },
]
`;
